cmd/bosun/sched/check.go

```diff
@@ -265,11 +265,34 @@ func (s *Schedule) CheckAlert(T miniprofiler.Timer, r *RunHistory, a *conf.Alert
 		}
 	}
 	unevalCount, unknownCount := markDependenciesUnevaluated(r.Events, deps, a.Name)
+	if err != nil {
+		removeUnknownEvents(r.Events, a.Name)
+	} else {
+		s.closeErrorEvent(r.Events, a.Name)
+	}
 
 	collect.Put("check.duration", opentsdb.TagSet{"name": a.Name}, time.Since(start).Seconds())
 	log.Printf("check alert %v done (%s): %v crits, %v warns, %v unevaluated, %v unknown", a.Name, time.Since(start), len(crits), len(warns), unevalCount, unknownCount)
 }
 
+func removeUnknownEvents(evs map[expr.AlertKey]*Event, alert string) {
```

Comment:
Why not accept a RunHistory for these two functions? I think that's a bit clearer, since then it's obvious that you aren't changing some other protected data structure.